autocheese.lua

```diff
@@ -0,0 +1,155 @@
+    elseif #unit.individual_drills > 0 then
+        return false
+    elseif unit.flags1.caged or unit.flags1.chained then
+        return false
+    elseif unit.military.squad_id ~= -1 then
+        local squad = df.squad.find(unit.military.squad_id)
+        -- this lookup should never fail
+        ---@diagnostic disable-next-line: need-check-nil
+        return #squad.orders == 0 and squad.activity == -1
+    end
+    return true
+end
+
+---find unit with a particular labor enabled
+---@param unit_labor df.unit_labor
+---@param job_skill df.job_skill
+---@param workshop df.building
+---@return df.unit|nil
+---@return integer|nil
+ function findAvailableLaborer(unit_labor, job_skill, workshop)
+    local max_unit = nil
+    local max_skill = -1
+    for _, unit in ipairs(dfhack.units.getCitizens(true, false)) do
+        if
+            unit.status.labors[unit_labor] and
+            unitIsAvailable(unit) and
+            ic.canAccessWorkshop(unit, workshop)
+        then
+            local unit_skill = dfhack.units.getNominalSkill(unit, job_skill, true)
+            if unit_skill > max_skill then
+                max_unit = unit
+                max_skill = unit_skill
+            end
+        end
+    end
+    return max_unit, max_skill
+end
+
+local function findMilkBarrel(min_liquids)
+    for _, container in ipairs(df.global.world.items.other.FOOD_STORAGE) do
+        if
+            not (container.flags.in_job or container.flags.forbid) and
+            container.flags.container and #container.general_refs >= min_liquids
+        then
+            local content_reference = dfhack.items.getGeneralRef(container, df.general_ref_type.CONTAINS_ITEM)
+            local contained_item = df.item.find(content_reference and content_reference.item_id or -1)
+            if contained_item then
+                local mat_info = dfhack.matinfo.decode(contained_item)
+                if mat_info:matches { milk = true } then
+                    return container
+                end
+            end
+        end
+    end
+end
+
+function findWorkshop()
+    for _,workshop in ipairs(df.global.world.buildings.other.WORKSHOP_FARMER) do
+        if
+            not workshop.profile.blocked_labors[df.unit_labor.MAKE_CHEESE] and
+            #workshop.jobs == 0 and #workshop.profile.permitted_workers == 0
+        then
+            return workshop
+        end
+    end
+end
+
+if dfhack_flags.module then
+    return
+end
+
+-- actual script action
+
+local argparse = require('argparse')
+
+local min_number = 50
+
+local _ = argparse.processArgsGetopt({...},
+{
+    { 'm', 'min-milk', hasArg = true,
+    handler = function(min)
+        min_number = argparse.nonnegativeInt(min, 'min-milk')
+    end }
+})
+
+
+local reagent = findMilkBarrel(min_number)
+
+if not reagent then
+    -- print('autocheese: no sufficiently full barrel found')
+    return
+end
+
+local workshop = findWorkshop()
+
+if not workshop then
+    print('autocheese: no workshop available')
```

Comment:
```suggestion
    print('autocheese: no Farmer's Workshop available')
```

Reply:
You successfully "tricked" me into blindly accepting this one. ALWAYS check for easy errors.  :roll_eyes: